### ISSUE TITLE
abstractstaticmethod false positive

### ISSUE THREAD

**Issue by vepain** [python/mypy#10356]
**Bug Report**

False positive error `Method must have at least one argument` for `abc.abstractstaticmethod` decorator.

**To Reproduce**

- MWE:
```python
from abc import abstractstaticmethod

class A():
    @abstractstaticmethod
    def statmeth():
        raise NotImplementedError
```

**Expected Behavior**

No error

**Actual Behavior**

`mypy(error): Method must have at least one argument`

**Environment**

- Mypy version used:
```python
mypy>=0.812<0.813
mypy-extensions>=0.4.3<0.5
```
- Mypy command-line flags:
```bash
mypy --config-file=tox.ini prog.py
```
- Mypy configuration options from `tox.ini` (and other config files):
```ini
[mypy]
ignore_missing_imports = True
follow_imports = silent
show_column_numbers = True
```
- Python version used: `python3.9`
- Operating system and version: `UBUNTU 20.04`

> Nota Bene: these are the versions I use, for some others the bug appears certainly

Thank you for your work !

**Comment by iFA88** [comment]
Confirm, same happens to me:
```python
from abc import ABCMeta, abstractstaticmethod

class T_Compressor(metaclass=ABCMeta):
	@abstractstaticmethod
	def compress(data:bytes) -> bytes: ...
```
Error message:`Self argument missing for a non-static method (or an invalid type for self)`
Environment:
- Python 3.9.2 (default, Feb 28 2021, 17:03:44)
- Debian 11.0

**Comment by iFA88** [comment]
Okay, its not mypy fault:
https://docs.python.org/3/library/abc.html#abc.abstractstaticmethod
`Deprecated since version 3.3:`
